solutions/pytorch_tests/Makefile

```diff
@@ -0,0 +1,135 @@
+TOP = $(abspath ../..)
+include $(TOP)/defs.mak
+
+APPBUILDER=$(TOP)/scripts/appbuilder
+
+DOCKER_REGISTRY           = docker.io
+DOCKER_ORG                = mystikos
+DOCKER_IMAGE              = pytorch-test
+DOCKER_FULL_NAME          = $(DOCKER_REGISTRY)/$(DOCKER_ORG)/$(DOCKER_IMAGE)
+DOCKER_FILE               = Dockerfile
+DOCKER_TAG                = $(PYTORCH_VERSION)
+
+BASE_IMAGE                = python:3.9-buster
+PYTORCH_VERSION           = v1.10.0
+BUILD_PROGRESS            = auto
+BUILD_TARGET              = test
+BUILD_ARGS                = --build-arg BASE_IMAGE=$(BASE_IMAGE) \
+							--build-arg PYTORCH_VERSION=$(PYTORCH_VERSION)
+EXTRA_OPTIONS             = --progress=$(BUILD_PROGRESS) \
+							--target $(BUILD_TARGET) \
+							-t $(DOCKER_FULL_NAME):$(DOCKER_TAG) \
+							$(BUILD_ARGS)
+
+DOCKER_BUILD              = DOCKER_BUILDKIT=1 \
+							docker build \
+							-f $(DOCKER_FILE) \
+							$(EXTRA_ARGS) .
+DOCKER_PUSH               = docker push $(DOCKER_FULL_NAME):$(DOCKER_TAG)
+DOCKER_RUN                = docker run --rm -it $(DOCKER_FULL_NAME):$(DOCKER_TAG)
+
+RUN_PYTEST                = $(MYST_EXEC) rootfs $(OPTS) --app-config-path config.json /usr/local/bin/python -m pytest -v
+
+ifdef STRACE
+OPTS += --strace
+endif
+
+.PHONY: all
+all: rootfs
+
+.PHONY: image
+image:
+	$(DOCKER_BUILD)
+
+.PHONY: docker-run
+docker-run:
+	$(DOCKER_RUN) /usr/local/bin/pytest /workspace/pytorch/test/test_autograd.py -v
+
+.PHONY: push
+push:
+	$(DOCKER_PUSH)
+
+appdir:
+	$(APPBUILDER) -i $(DOCKER_FULL_NAME):$(DOCKER_TAG) -v
+
+rootfs: appdir
+	$(MYST) mkext2 appdir rootfs
+
+.PHONY: run-test-autograd
+run-test-autograd: rootfs
+	$(RUN_PYTEST) /workspace/pytorch/test/test_autograd.py -k "not test_deep_reentrant and not test_no_grad_copy_sparse and not test_profiler_seq_nr and not test_record_function and not test_thread_shutdown" 2>&1 | tee $@.out
+	grep -q -E '400 passed.*68 skipped.*7 deselected.*1 xfailed' $@.out
```

Comment:
Is there a way to search for the negative feedback such as "failed" instead of the positive feedbacks?

I am trying to avoid the need to frequently update the numbers for "passed" or "skipped" in this Makefile.

Reply:
I also thought about making the passing condition less strict to avoid false alarm. pytest cmd will return a non-zero code if some tests fail. But I'm curious if these number would ever change, e.g. some passed cases will become skipped or deselected because of Mystikos code change. That's why I added the grep here. So that at least we are notified. How about we keep the strict condition for now until these is evidence that these numbers are not stable?